sdk/core/azure-core-serializeradapter-jackson/src/main/java/com/azure/core/serializeradapter/jackson/JacksonAdapter.java

```diff
@@ -0,0 +1,290 @@
+    }
+
+    /**
+     * An instance of {@link ObjectMapperShim} to serialize/deserialize objects.
+     */
+    private final ObjectMapperShim mapper;
+
+    private final ObjectMapperShim xmlMapper;
+
+    private final ObjectMapperShim headerMapper;
+
+    /**
+     * Raw mappers are needed only to support deprecated simpleMapper() and serializer().
+     */
+    private ObjectMapper rawOuterMapper;
+    private ObjectMapper rawInnerMapper;
+
+    /**
+     * Creates a new JacksonAdapter instance with default mapper settings.
+     */
+    public JacksonAdapter() {
+        this((outerMapper, innerMapper) -> {
+        });
+    }
+
+    /**
+     * Creates a new JacksonAdapter instance with Azure Core mapper settings and applies additional configuration
+     * through {@code configureSerialization} callback.
+     *
+     * {@code configureSerialization} callback provides outer and inner instances of {@link ObjectMapper}. Both of them
+     * are pre-configured for Azure serialization needs, but only outer mapper capable of flattening and populating
+     * additionalProperties. Outer mapper is used by {@code JacksonAdapter} for all serialization needs.
+     *
+     * Register modules on the outer instance to add custom (de)serializers similar to {@code new JacksonAdapter((outer,
+     * inner) -> outer.registerModule(new MyModule()))}
+     *
+     * Use inner mapper for chaining serialization logic in your (de)serializers.
+     *
+     * @param configureSerialization Applies additional configuration to outer mapper using inner mapper for module
+     * chaining.
+     */
+    public JacksonAdapter(BiConsumer<ObjectMapper, ObjectMapper> configureSerialization) {
+        Objects.requireNonNull(configureSerialization, "'configureSerialization' cannot be null.");
+        this.headerMapper = ObjectMapperShim.createHeaderMapper();
+        this.xmlMapper = ObjectMapperShim.createXmlMapper();
+        this.mapper = ObjectMapperShim.createJsonMapper(ObjectMapperShim.createSimpleMapper(),
+            (outerMapper, innerMapper) -> captureRawMappersAndConfigure(outerMapper, innerMapper, configureSerialization));
+    }
+
+    /**
+     * Temporary way to capture raw ObjectMapper instances, allows to support deprecated simpleMapper() and
+     * serializer()
+     */
+    private void captureRawMappersAndConfigure(ObjectMapper outerMapper, ObjectMapper innerMapper, BiConsumer<ObjectMapper, ObjectMapper> configure) {
+        this.rawOuterMapper = outerMapper;
+        this.rawInnerMapper = innerMapper;
+
+        configure.accept(outerMapper, innerMapper);
+    }
+
+    /**
+     * Gets a static instance of {@link ObjectMapper} that doesn't handle flattening.
+     *
+     * @return an instance of {@link ObjectMapper}.
+     * @deprecated deprecated, use {@code JacksonAdapter(BiConsumer<ObjectMapper, ObjectMapper>)} constructor to
+     * configure modules.
+     */
+    @Deprecated
+    protected ObjectMapper simpleMapper() {
```

Comment:
can we clean this one up and remove all APIs that use `ObjectMapper` directly? We can use shim here.

Onboarding onto this package would still require some work, and it would be good opportunity to clean up any raw ObjectMapper usages